desktop/components/react/utils/template_renderer.js

```diff
@@ -0,0 +1,41 @@
+import invariant from 'invariant'
+import jade from 'jade'
+import path from 'path'
+
+import {
+  isArray,
+  isString
+} from 'underscore'
+
+export default function templateRenderer (templates, options = {}) {
+  const isValid = isArray(templates) || isString(templates)
+
+  invariant(isValid,
+    '(lib/template_renderer.js) ' +
+    'Error rendering templates: `templates` must be a string or array of ' +
+    'strings representing the path to the template'
+  )
+
+  templates = isArray(templates)
+    ? templates
+    : [templates]
+
+  const basePath = options.basePath || ''
+
+  const templateFns = templates.map(file => {
+    return jade.compileFile(path.join(basePath, file), {
+      cache: true,
+      ...options
+    })
+  })
+
+  return {
```

Comment:
Following a few different conventions, I like the idea of a keeping rendering lazy, and being able to preload a bunch of templates for variable data sets that are later called with `render`.